cpp/src/parquet/page_index_test.cc

```diff
@@ -0,0 +1,208 @@
+  const std::vector<int64_t> max_values = {90, 90, 80, 70};
+
+  TestReadTypedColumnIndex<Int64Type>(column_id, num_pages, boundary_order, page_indices,
+                                      null_pages, min_values, max_values, has_null_counts,
+                                      null_counts);
+}
+
+TEST(PageIndex, ReadDoubleColumnIndex) {
+  const int column_id = 7;
+  const size_t num_pages = 528;
+  const BoundaryOrder::type boundary_order = BoundaryOrder::Unordered;
+  const std::vector<size_t> page_indices = {0, 51, 212, 527};
+  const std::vector<bool> null_pages = {false, false, false, false};
+  const bool has_null_counts = true;
+  const std::vector<int64_t> null_counts = {0, 0, 0, 0};
+  const std::vector<double> min_values = {-0, 30.3, 10.1, 40.4};
+  const std::vector<double> max_values = {90.9, 90.9, 90.9, 60.6};
+
+  TestReadTypedColumnIndex<DoubleType>(column_id, num_pages, boundary_order, page_indices,
+                                       null_pages, min_values, max_values,
+                                       has_null_counts, null_counts);
+}
+
+TEST(PageIndex, ByteArrayColumnIndex) {
+  const int column_id = 9;
+  const size_t num_pages = 352;
+  const BoundaryOrder::type boundary_order = BoundaryOrder::Ascending;
+  const std::vector<size_t> page_indices = {0, 128, 256};
+  const std::vector<bool> null_pages = {false, false, false};
+  const bool has_null_counts = true;
+  const std::vector<int64_t> null_counts = {0, 0, 0};
+
+  // All min values are "0" and max values are "9".
+  const std::string_view min_value = "0";
+  const std::string_view max_value = "9";
+  const std::vector<ByteArray> min_values = {ByteArray{min_value}, ByteArray{min_value},
+                                             ByteArray{min_value}};
+  const std::vector<ByteArray> max_values = {ByteArray{max_value}, ByteArray{max_value},
+                                             ByteArray{max_value}};
+
+  TestReadTypedColumnIndex<ByteArrayType>(column_id, num_pages, boundary_order,
+                                          page_indices, null_pages, min_values,
+                                          max_values, has_null_counts, null_counts);
+}
+
+TEST(PageIndex, ReadBoolColumnIndex) {
+  const int column_id = 1;
+  const size_t num_pages = 82;
+  const BoundaryOrder::type boundary_order = BoundaryOrder::Ascending;
+  const std::vector<size_t> page_indices = {0, 16, 64};
+  const std::vector<bool> null_pages = {false, false, false};
+  const bool has_null_counts = true;
+  const std::vector<int64_t> null_counts = {0, 0, 0};
+  const std::vector<bool> min_values = {false, false, false};
+  const std::vector<bool> max_values = {true, true, true};
+
+  TestReadTypedColumnIndex<BooleanType>(column_id, num_pages, boundary_order,
+                                        page_indices, null_pages, min_values, max_values,
+                                        has_null_counts, null_counts);
+}
+
```

Comment:
Is there also a fixed-length byte array column that we can test?

Reply:
You can generate such a file with another writer and add it to `parquet-testing` perhaps? That would ensure we're compatible with other implementations (mere roundtripping doesn't guarantee that).

Reply:
I did some research and found that FIXED_LENGTH_BYTE_ARRAY type is only used for `interval` and `decimal` logical types. For `interval` type, statistics is disabled by the spec. For `decimal` type, `INT32`, `INT64` and `BYTE_ARRAY` physical types are used by default in the `parquet-mr`, I cannot simply use `parquet-cli` command line to generate the file. Will look into the test cases to find some examples.

Reply:
Added test case `FixedLengthByteArrayColumnIndex`